TSRM/tsrm_win32.c

```diff
@@ -662,6 +665,19 @@ TSRM_API int shmget(key_t key, size_t size, int flags)
 		}
 	}
 
+	if (key == IPC_PRIVATE) {
+		/* This should call shm_get with a brand new key id that isn't used yet. See https://man7.org/linux/man-pages/man2/shmget.2.html
+		 * Because shmop_open can be used in userland to attach to shared memory segments, use high positive numbers to avoid accidentally conflicting with userland. */
+		key = (php_rand() & 0x3fffffff) + 0x40000000;
```

Comment:
I assume I probably shouldn't be depending on ext/ (ext/standard/php_rand.h) from TSRM (similar rules to Zend?), but I'm not completely sure of that. Other uses are just for defined constants

Maybe secure random byte generation could be moved to Zend

php_random_int would be unaffected by calls to mt_srand


```c
		if (php_random_int(0x4000000, 0x7fff0000, &result, 0) == FAILURE) {
			key++;
		} else {
			key = result;
		}
```